metamorph/processor.go

```diff
@@ -312,6 +313,17 @@ func (p *Processor) LoadUnmined() {
 	defer span.Finish()
 
 	err := p.store.GetUnmined(spanCtx, func(record *store.StoreData) {
+
+		if p.now().Sub(record.StoredAt) > txDeletionPeriod {
```

Comment:
1. do we really need `time.Now` as another field in processor?
2. would it be more readable to extract this condition as function:

```
func isExpired(record *store.StoreData) bool {
  return time.Now().Sub(record.StoredAt) > txDeletonPeriod
}
```

and then 

```
if isExpired(record) {
 // handle deletion
}
``` 
What do you think?

Reply:
1. The field was added as `now: func() time.Time` so that you can set this function. By default the function is simply `time.Now()` but when you do tests you may want to control what time is returned. Then you can set it to `func() time.Time { return time.Date(...)}`. That is why it is a field
2. Agree. Will see how to best put it in a function